src/libstd/io/buffered.rs

```diff
@@ -449,12 +563,47 @@ impl<W: Write> LineWriter<W> {
     ///
     /// Caution must be taken when calling methods on the mutable reference
     /// returned as extra writes could corrupt the output stream.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs::File;
+    /// use std::io::prelude::*;
+    /// use std::io::LineWriter;
+    ///
+    /// # fn foo() -> std::io::Result<()> {
+    /// let file = try!(File::create("poem.txt"));
+    /// let mut file = LineWriter::new(file);
+    ///
+    /// let reference = file.get_mut();
+    ///
+    /// // we can use reference just like file
+    /// try!(reference.write(b"Mary had a little lamb, its face as white as snow."));
```

Comment:
Similarly to the other `get_mut` method on `BufWriter`, this may be a bit odd to have an example doing so when it's warned about as "you shouldn't do this" above.